src/libServer/LookupServer.cpp

```diff
@@ -627,6 +632,47 @@ Json::Value LookupServer::GetTransaction(const string& transactionHash) {
   }
 }
 
+Json::Value LookupServer::GetSoftConfirmedTransaction(const string& txnHash) {
+  LOG_MARKER();
+
+  if (!LOOKUP_NODE_MODE) {
+    throw JsonRpcException(RPC_INVALID_REQUEST, "Sent to a non-lookup");
+  }
+
+  try {
+    TxBodySharedPtr tptr;
+    TxnHash tranHash(txnHash);
+    if (txnHash.size() != TRAN_HASH_SIZE * 2) {
+      throw JsonRpcException(RPC_INVALID_PARAMS, "Size not appropriate");
+    }
+    bool isPresent = BlockStorage::GetBlockStorage().GetTxBody(tranHash, tptr);
+    bool isPresentHistorical = false;
```

Comment:
Maybe cross-checking against historical DB is not necessary? Also, this db is for diagnostic purposes, so using it as a canonical source might not be a good idea.

Reply:
noted